src/app/GitUI/CommandsDialogs/FormClone.cs

```diff
@@ -55,6 +55,12 @@ protected override void OnRuntimeLoad(EventArgs e)
             _NO_TRANSLATE_From.DataSource = repositoryHistory;
             _NO_TRANSLATE_From.DisplayMember = nameof(Repository.Path);
 
+            IList<Repository> localsHistory = ThreadHelper.JoinableTaskFactory.Run(RepositoryHistoryManager.Locals.LoadRecentHistoryAsync);
```

Comment:
What is a typical delay for this?

Reply:
@gerhardol this should be consistent with what we already have with the repo dropdown in the main form. Given this is a standalong dialog, this shouldn't be an issue at all.

Reply:
The repo dropdown menu is unusable for me, but I have 150 repos (I would like to use more).
I have privately removed branches there to have something working (I still want branches in the dashboard).
It may be OK here though.

Reply:
> The repo dropdown menu is unusable for me, but I have 150 repos (I would like to use more). I have privately removed branches there to have something working (I still want branches in the dashboard). It may be OK here though.

This is an existing implementation, and in this dialog it shouldn't be a big deal. And we can certainly entertain an option to make this functionality more performant, if possible (e.g., not loading branches).